addons/account/models/account_move_line.py

```diff
@@ -2491,7 +2491,7 @@ def _prepare_analytic_lines(self):
 
             for account_id, distribution in self.analytic_distribution.items():
                 line_values = self._prepare_analytic_distribution_line(float(distribution), account_id, distribution_on_each_plan)
-                if not float_is_zero(line_values.get("amount"), precision_digits=self.env.company.currency_id.decimal_places):
+                if not float_is_zero(line_values.get("amount"), precision_digits=self.currency_id.decimal_places):
```

Comment:
```suggestion
                if not self.currency_id.is_zero(line_values.get("amount")):
```

The you might want to remove the import of `float_is_zero` also